classes/security/UserGroupDAO.inc.php

```diff
@@ -458,6 +459,33 @@ public function getByContextId($contextId = null, $dbResultRange = null)
         return new DAOResultFactory($result, $this, '_returnFromRow', [], $sql, $params, $dbResultRange);
     }
 
+    /**
+     * Retrieve user groups counts
+     */
+    public function getUserCountByContextId(int $contextId = null): array
+    {
+        $query = DB::table('user_groups', 'ug')
+            ->join('user_user_groups AS uug', 'uug.user_id', '=', 'ug.user_group_id')
+            ->whereColumn('uug.user_group_id', '=', 'ug.user_group_id')
+            ->join('users AS u', 'u.user_id', '=', 'uug.user_id')
+            ->when($contextId, function (Builder $query) use ($contextId): void {
+                $query->where('ug.context_id', '=', $contextId);
+            })
+            ->where('u.disabled', '=', 0)
+            ->groupBy('ug.user_group_id')
+            ->select('ug.user_group_id')
+            ->selectRaw('COUNT(0) AS count');
+
+        $sql = $query->toSql();
+        $params = $query->getBindings();
+
+        $result = [];
+        foreach ($this->retrieveRange($sql, $params) as $row) {
```

Comment:
Is it necessary to use `retrieveRange`? It looks to me like you can just use a `->select(...)->get()` and get rid of `$sql`, `$params`, etc.

Reply:
Copy/paste failure 😑